services/dsp-x/src/index.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ Copyright 2022 Google LLC
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+// DSP
+import express, {Application, Request, Response} from 'express';
+import {buildInterestGroup} from './server/helper/build-interest-group.js';
+
+const {EXTERNAL_PORT, PORT, DSP_X_HOST, DSP_X_TOKEN, DSP_X_DETAIL, SHOP_HOST} =
+  process.env;
+
+const app: Application = express();
+
+app.use((req: Request, res: Response, next) => {
+  res.setHeader('Access-Control-Allow-Origin', '*');
+  res.setHeader('Origin-Trial', DSP_X_TOKEN as string);
+  next();
+});
+
+app.use(express.urlencoded({extended: true}));
+app.use(express.json());
+
+app.use(
+  express.static('src/public', {
+    setHeaders: (res: Response, path) => {
+      const url: URL = new URL(path, `https://${DSP_X_HOST}`);
+      if (url.pathname.endsWith('bidding-logic.js')) {
+        return res.set('X-Allow-FLEDGE', 'true');
+      }
+    },
+  }),
+);
+
+app.use((req: Request, res: Response, next) => {
+  if (req.get('sec-fetch-dest') === 'fencedframe') {
+    res.setHeader('Supports-Loading-Mode', 'fenced-frame');
+  }
+  next();
+});
+
+app.set('view engine', 'ejs');
+app.set('views', 'src/views');
+
+app.get('/', async (req: Request, res: Response) => {
+  const title = DSP_X_DETAIL;
+  res.render('index', {title, DSP_X_HOST, SHOP_HOST, EXTERNAL_PORT});
+});
+
+app.get(
+  '/uc-:useCaseName/join-ad-interest-group.html',
+  async (req: Request, res: Response) => {
+    const {useCaseName} = req.params;
+    const title = 'DSP X - Join Ad Interest Group';
+    res.render(`uc-${useCaseName}/join-ad-interest-group`, {
```

Comment:
Do we need the use case name to be a user provided value? I think it would be better to replace this with `uc-ba` for now since this is the only use case relevant for DSP X and Y